index/index_monitor.go

```diff
@@ -0,0 +1,46 @@
+package index
+
+// helpers to atomically switch from using one index to another
+
+import (
+	"sync"
+)
+
+type IndexMonitor struct {
```

Comment:
Can we just put this functionality into `sequins`, or do we need it separate for testing? it seems a bit heavyweight out on its own.